File: clients/node/lib/encryptors/aes-256-cbc.js
const crypto = require('crypto'),
  algorithm = 'AES-256-CBC';

async function encrypt(value, password, iv) {
  const cipher = crypto.createCipheriv(algorithm, password, iv);
  let encrypted = cipher.update(value, 'utf8', 'hex')
  encrypted += cipher.final('hex');
  return [ encrypted, iv ];
}

async function decrypt(encrypted, password) {

  const content = encrypted[0];
  const iv = encrypted[1];

  const decipher = crypto.createDecipheriv(algorithm, password, iv);
  let dec = decipher.update(content, 'hex', 'utf8');
  dec += decipher.final('utf8');
  return dec;
}

module.exports = {
  encrypt,
  decrypt
};

File: clients/ruby/lib/ElencyConfig/AES256CBC.rb
require 'base64'
require 'openssl'

class AES256CBC

  def encrypt(value, password, iv)
    aes = ::OpenSSL::Cipher::Cipher.new("AES-256-CBC")
    aes.padding = 1
    aes.encrypt
    aes.key = password
    aes.iv = iv
    encrypted = aes.update(value) + aes.final
    hexed = encrypted.unpack('H*')[0]
    return [hexed, iv]
  end

  def decrypt(encrypted, password)
    aes = ::OpenSSL::Cipher::Cipher.new("AES-256-CBC")
    aes.padding = 1
    aes.decrypt
    aes.key = password
    aes.iv = encrypted[1]

    dehexed = [encrypted[0]].pack('H*')
    decrypted = aes.update(dehexed) + aes.final
    return decrypted
  end

end